### PR TITLE
Feat/more workflow wrapper extensions 2

### PR DESCRIPTION
**Pull request by seanpearsonuk**
Update to the workflow interface. Details are in the extended usage in test_meshing_workflow.py.

### CODE REVIEW

File: tests/test_meshing_workflow.py [hunk 1]
```diff
@@ -590,11 +579,166 @@ def downstream_names(task):
...
+    )
+    create_volume_mesh = watertight.create_volume_mesh
+    assert create_volume_mesh is not None
+    watertight.DeleteTasks(ListOfTasks=["Generate the Volume Mesh"])
+    # I assume that what's going on here is that due to DeleteTasks we are triggering
+    # change events in the server but those events are (still) being transmitted after
+    # DeleteTasks has returned. Hence, the dynamic watertight Python interface
+    # is still updating after the command has returned and the client can try to access
+    # while it is in that update phase, leading to (difficult to understand) exceptions.
+    # Temporarily sleeping in the test. I note that the core event tests use sleeps also.
```
Comment (seanpearsonuk) [inline tests/test_meshing_workflow.py:704]: this doesn't seem to be sleeping in the latest version

File: src/ansys/fluent/core/workflow.py [hunk 2]
```diff
@@ -327,6 +655,21 @@ def __call__(self):
         """Delegate calls to the underlying workflow."""
         return self._workflow()
 
+    def _wait_on_refresh(self, time_unit=0.1, skip_check=False):
+        if not skip_check:
+            t0 = time()
+        if skip_check or threading.get_ident() == self._main_thread_ident:
+            refresh_count = self._refresh_count
+            orig_refresh_count = refresh_count
+            sleep(20 * time_unit)
```
Comment (seanpearsonuk) [inline src/ansys/fluent/core/workflow.py:664]: Here is the sleep that's not in the test

**Review by hpohekar — APPROVED** [review]

**Review by prmukherj — APPROVED** [review]

File: src/ansys/fluent/core/services/datamodel_se.py [hunk 3]
```diff
@@ -1137,7 +1143,7 @@ def __del__(self):
         request.commandid = self.path[-1][1]
         try:
             self.service.delete_command_arguments(request)
-        except ValueError:
+        except BaseException:
             # "Cannot invoke RPC on closed channel!"
             pass
```
Comment (raph-luc) [inline src/ansys/fluent/core/services/datamodel_se.py:1148]:
```suggestion
        except BaseException as exc:
            logger.info("__del__ %s: %s" % (type(exc).__name__, exc))
```
Removing the commented line as it is no longer applicable, and adding a logging report of the BaseException that was caught (particularly as this is a very general exception catch).

File: src/ansys/fluent/core/session_base_meshing.py [hunk 4]
```diff
@@ -3,11 +3,13 @@
 
 from ansys.fluent.core.fluent_connection import FluentConnection
 from ansys.fluent.core.meshing.meshing import Meshing
+
+# from ansys.fluent.core.workflow import WorkflowWrapper
```
Comment (raph-luc) [inline src/ansys/fluent/core/session_base_meshing.py:7]:
```suggestion
```

File: src/ansys/fluent/core/workflow.py [hunk 5]
```diff
@@ -344,6 +687,40 @@ def _task_by_id(self, task_id):
...
+        try:
+            result = getattr(self._workflow, attr)
+            if result:
+                return result
+        except AttributeError:
+            pass
+
+    # def _task_with_cmd_matching_help_string(self, help_string):
+    #    init_task_accessors(self)
+    #    return getattr(self, help_string)
```
Comment (raph-luc) [inline src/ansys/fluent/core/workflow.py:700]:
```suggestion
```

File: tests/test_datamodel_service.py [hunk 6]
```diff
@@ -143,6 +143,41 @@ def test_add_on_affected(new_mesh_session):
...
+    assert calls == 3 * [True]
+    execute_state = [meshing.workflow()]
+    import_geom.Execute()
+    calls_after_execute = []
+    loop_count = 10
+    for i in range(loop_count):
+        sleep(5)
+        calls_after_execute.append(list(calls))
+        execute_state.append(meshing.workflow())
+    # assert execute_state[1] != execute_state[0]
```
Comment (raph-luc) [inline tests/test_datamodel_service.py:174]:
```suggestion
```

File: tests/test_meshing_workflow.py [hunk 7]
```diff
@@ -590,11 +579,166 @@ def downstream_names(task):
...
+        == 1
+    )
+    watertight.DeleteTasks(ListOfTasks=["Create Regions"])
+    assert watertight.describe_geometry.create_regions is None
+    assert watertight.describe_geometry.enclose_fluid_regions
+    watertight.describe_geometry.enclose_fluid_regions.delete()
+    assert watertight.describe_geometry.enclose_fluid_regions is None
+    watertight.create_volume_mesh.delete()
+    assert watertight.create_volume_mesh is None
+    # watertight.insert_new_task()
```
Comment (raph-luc) [inline tests/test_meshing_workflow.py:721]:
```suggestion
```

File: tests/test_datamodel_service.py [hunk 8]
```diff
@@ -143,6 +143,41 @@ def test_add_on_affected(new_mesh_session):
...
+    assert calls == 2 * [True]
+    import_geom.Arguments = {"FileName": geom}
+    sleep(1)
+    assert calls == 3 * [True]
+    execute_state = [meshing.workflow()]
+    import_geom.Execute()
+    calls_after_execute = []
+    loop_count = 10
+    for i in range(loop_count):
+        sleep(5)
```
Comment (raph-luc) [inline tests/test_datamodel_service.py:171]: Are there any bools or objects that we could check here to minimize the amount of time wasted sleeping? Perhaps check if a new `bool` has been added to the `calls` list, or until list length has increased? Could perhaps use one of these for convenience after they are merged: https://github.com/ansys/pyfluent/pull/1748#discussion_r1252256494

**Review by mkundu1 — APPROVED** [review]

**Pull request merged**